obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    Returns an obspy.core.event.Catalog type: note this changed for version \
+    0.1.0 from the inbuilt class types.
+
+    :type sfile: str
+    :param sfile: Path to the s-file
+
+    :returns: :class: obspy.core.event.Event
+    """
+    with open(sfile, 'r') as f:
+        header = _readheader(f=f)
+    return header
+
+
+def _readheader(f):
+    """Internal header reader.
+    :type f: file
+    :param f: File open in read-mode.
+
+    :returns: obspy.core.event.Event"""
+    f.seek(0)
+    # Base populate to allow for empty parts of file
+    new_event = Event()
+    topline = f.readline()
+    if not len(topline.rstrip()) == 80:
+        f.close()
+        raise IOError('s-file has a corrupt header, not 80 char long')
+    f.seek(0)
+    for line in f:
+        if line[79] in [' ', '1']:
+            topline = line
+            break
+        if line[79] == '7':
+            f.close()
+            raise IOError('No header found, corrupt s-file?')
+    try:
+        sfile_seconds = int(topline[16:18])
+        if sfile_seconds == 60:
+            sfile_seconds = 0
+            add_seconds = 60
+        else:
+            add_seconds = 0
+        new_event.origins.append(Origin())
+        new_event.origins[0].time = UTCDateTime(int(topline[1:5]),
+                                                int(topline[6:8]),
+                                                int(topline[8:10]),
+                                                int(topline[11:13]),
+                                                int(topline[13:15]),
+                                                sfile_seconds,
+                                                int(topline[19:20]) *
+                                                100000)\
+            + add_seconds
+    except:
+        f.close()
+        IOError("Couldn't read a date from sfile")
+    # new_event.loc_mod_ind=topline[20]
+    new_event.event_descriptions.append(EventDescription())
+    new_event.event_descriptions[0].text = topline[21:23]
+    # new_event.ev_id=topline[22]
+    if not _float_conv(topline[23:30]) == 999:
+        new_event.origins[0].latitude = _float_conv(topline[23:30])
+        new_event.origins[0].longitude = _float_conv(topline[31:38])
+        new_event.origins[0].depth = _float_conv(topline[39:43]) * 1000
+    else:
+        # The origin 'requires' a lat & long
+        new_event.origins[0].latitude = float('NaN')
+        new_event.origins[0].longitude = float('NaN')
+        new_event.origins[0].depth = float('NaN')
+    # new_event.depth_ind = topline[44]
+    # new_event.loc_ind = topline[45]
+    new_event.creation_info = CreationInfo(agency_id=topline[45:48].
+                                           strip())
+    ksta = Comment(text='Number of stations=' +
+                   topline[49:51].strip())
+    new_event.origins[0].comments.append(ksta)
+    # new_event.origins[0].nsta??? = _int_conv(topline[49:51])
+    if not _float_conv(topline[51:55]) == 999:
```

Comment:
See, `_float_conv` is making you do double  the conversions, whereas a simple `try:/except ValueError: pass` would have been simpler.